### ISSUE TITLE
aws-amplify/auth  fetchAuthSession makes network calls to https://cognito-identity.us-east-1.amazonaws.com/ even after setting up custom domain on cognito pool

### ISSUE THREAD

**Issue by madCodes** [aws-amplify/amplify-js#13445]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Authentication

### Amplify Version

v6

### Amplify Categories

auth

### Backend

Other

### Environment information

<details>

```
# Put output below this line


```

</details>

### Describe the bug

I have importing existing Cognito resource using amplify cli and set up authentication using react-ui package in nextjs. Everything runs smooth, but i need to change the domain which amplify cli uses for auth calls like  fetchAuthSession. I have added custom domain to cognito using amazon web console and tried to pull cloud changes hoping to see "oauth": {"domain":"my-custom-domain.com"}. It never synced. I forcefully updated while setting Amplify.configure like below.

```
amplifyconfig.oauth.domain = "auth.mydomain.com";

Amplify.configure(amplifyconfig, {
  ssr: false,
});

```

No matter what i do, the amplify sdk always hit "cognito-identity.us-east-1.amazonaws.com" domain. Not sure, what am i missing here. Thanks.

### Expected behavior

Hit my custom domain for all the amplify auth calls

### Reproduction steps

Import existing cognito without custom domain i nto amplify. 
Set the custom domain outside amplify, may be on aws web console. 
Try configuring custom domain in the amplify

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by madCodes** [comment]
@cwomack Were you able to reproduce this? Let me know if you need any further details. This is a show stopper for us as we need to run our app in a restricted environment where only my app domains are whitelisted and people are not able to login because amplify sdk hits  *.amazonaws.com domains.

**Comment by cwomack** [comment]
@madCodes, appreciate your patience with my response.  While we support custom user pool endpoints, we do not currently provide any input for custom domains using identity pools with Amplify.  We've marked this as a feature request and will review it internally.

**Comment by cwomack** [comment]
Following up on this further, @madCodes.  Would using Custom User Pool endpoints work for your use case instead?  I may have misunderstood the intent within this issue, but based on the related Discord thread I'm wondering if this Custom User Pool endpoints would work instead.  These docs aren't very detailed, but the [userPoolEndpoint](https://aws-amplify.github.io/amplify-js/api/interfaces/aws_amplify.adapter_core._Reference_Types_.CognitoUserPoolConfig.html#userPoolEndpoint) docs may help here to see that it's an available option in the `CognitoUserPoolConfig` interface.

**Comment by madCodes** [comment]
@cwomack We are open to any solution which will avoid amplify auth hitting aws domains.If i understand correctly,  i should pass "auth.mydomain.com" to userPoolEndpoint field? Sorry, but could you please help on how to pass this ?

```
import amplifyconfig from "@/amplifyconfiguration.json";
Amplify.configure(amplifyconfig);
```


Content of the amplifyconfiguration.json is as follows.


`{
  "aws_project_region": "us-east-1",
  "aws_cognito_region": "us-east-1",
  "aws_user_pools_id": "us-east-1_XXXXXX",
  "aws_user_pools_web_client_id": "1gc2XXXXXX",
  "oauth": {
    "domain": "auth.mydomain.com"
  },
  "aws_cognito_username_attributes": ["EMAIL"],
  "aws_cognito_social_providers": [],
  "aws_cognito_signup_attributes": ["EMAIL"],
  "aws_cognito_mfa_configuration": "OFF",
  "aws_cognito_mfa_types": [],
  "aws_cognito_password_protection_settings": {
    "passwordPolicyMinLength": 8,
    "passwordPolicyCharacters": []
  },
  "aws_cognito_verification_mechanisms": ["EMAIL"]
}
`

**Comment by israx** [comment]
Hello @madCodes. Amplify makes API requests to both user-pool and identity-pool Cognito endpoints. Bellow I added more context for both.

1. Override default user-pool endpoint, e.g. `https://cognito-idp`

You need to configure a custom endpoint via the Cognito console and configuring Amplify as follows:

```
Auth: {
    Cognito: {
      userPoolId: "xxxxxx",
      userPoolClientId: "xxxxx",
      identityPoolId: "xxxx",
      userPoolEndpoint:'endpoint', 
 }
}
```

2. Override default identity-pool endpoint, e.g.  
`https://cognito-identity`

Unfortunately Amplify doesn't support overriding the default identity-pool endpoint

**Comment by Neuroforge** [comment]
This is so painful. I managed to get the signIn request after upgrading from v5 - v6 and then it failed when fetching the session. Is there any way to patch this?

**Comment by jglee96** [comment]
Could be related with #13552

**Comment by Neuroforge** [comment]
> Could be related with #13552

Oh man! This is just sitting there. Are there any ways to get his moving forward?

https://github.com/aws-amplify/amplify-js/pull/13552

**Comment by cwomack** [comment]
@Neuroforge, @jglee96, and @madCodes

Just wanted to ping you all to say that we are indeed tracking the PR for this feature in #13552 and that will be further reviewed/worked on once we get the bug resolved that is detailed within #13650.

**Comment by Locaided** [comment]
Hello,
is there any update on this? I see that the linked PRs have been closed.

**Comment by osama-rizk** [comment]
Yes, a change has been merged and released to support configuring  `identityPoolEndpoint`. Here is how you can do so.

```

  Amplify.configure({
    Auth: {
      Cognito: {
        identityPoolEndpoint: "https://example.com"
      }
    }
  })

```
Please make sure to use version > aws-amplify@6.15.0 here is the [release details](https://github.com/aws-amplify/amplify-js/releases/tag/aws-amplify%406.15.0)
